### PR TITLE
Add additional information to note for referencing packed field diagnostic

### PR DESCRIPTION
**Pull request by bindsdev**
Addresses #110199

### CODE REVIEW

**Comment by TaKO8Ki** [comment]
Could you fix some unit test stderr files?

File: compiler/rustc_mir_transform/src/check_packed_ref.rs [hunk 1]
```diff
@@ -56,8 +56,9 @@ impl<'tcx> Visitor<'tcx> for PackedRefChecker<'_, 'tcx> {
                         "reference to packed field is unaligned"
                     )
                     .note(
```
Comment (compiler-errors) [inline compiler/rustc_mir_transform/src/check_packed_ref.rs:58]: This should probably be broken up into several notes, e.g.

> packed structs are only aligned by one byte, and many modern architectures penalize unaligned field accesses

> creating a misaligned reference is undefined behavior (even if that ...

Comment (compiler-errors) [reply compiler/rustc_mir_transform/src/check_packed_ref.rs:58]: I tried to phrase my suggestion to avoid these periods in the note, because they don't match the style of other notes. Can you rephrase this to be more like what I suggested here?

**Review by compiler-errors — CHANGES_REQUESTED** [review]
Please see my follow-up comment about the phrasing of the note message. Also, Merge commits are not allowed in PRs, and this really should be squashed into one commit.

File: compiler/rustc_mir_transform/src/check_packed_ref.rs (hunk 1 above)
Comment (bindsdev) [reply compiler/rustc_mir_transform/src/check_packed_ref.rs:58]: Oh yeah, I totally missed that one. Apologies.

**Comment by bindsdev** [comment]
I messed something up. I am going to reopen the PR.
